### ISSUE TITLE
Editor shows old script when using ResourceSaver in tool mode. 

### ISSUE THREAD

**Issue by AlexHolly** [godotengine/godot#5501]
After using "ResourceSaver" with tool mode, the editor is not reloading my edited script.

Example project:
[BUG.zip](https://github.com/godotengine/godot/files/341406/BUG.zip)

Open the project with godot. Go to the project folder with your filesystem open ground.gd and remove the last line "export var id=-1"

1. Open the main.scn and the tileset.tscn in godot
2. See that the child "Ground" in the tileset.tscn has a script where "export var id=-1" is missing.
3. Go to main.scn and click on the exported "Update" property to trigger the ground.gd update.
4. Go to your filesystem and see that "export var id=-1" was added to ground.gd.
5. Go to the "ground.gd" in the editor and see that nothing has changed.
6. Repopen the tileset.tscn has no effect.
7. Restart the project shows now the updated  "ground.gd"
8. It is also possible to see the update without Reopen or Restart when you edit the project folder with your filesystem(delete or add a file).

**Comment by reduz** [comment]
well, this is not exactly a bug, it's simply that the script editor is not expecting you to change the scripts from a tool. Maybe an API to refresh the edited script would work?

**Comment by AlexHolly** [comment]
Yes, I have no problem with executing a function to tell the editor that a script was changes.

**Comment by Calinou** [comment]
@AlexHolly Is this issue still relevant in the `master` branch?

**Comment by AlexHolly** [comment]
Still not working 

Tested with:
![image](https://user-images.githubusercontent.com/4741886/63135675-fe2c6000-bfce-11e9-87cd-a023d483f2a8.png)

Project update
[bug.zip](https://github.com/godotengine/godot/files/3507627/bug.zip)

**Comment by KoBeWi** [comment]
There is an editor setting called `auto reload scripts on external change`, which makes the script reload on external change. But seems like this case is not recognized as external. I opened the script in VS Code. Changes from `tool` will be reflected in VS Code, but not in the built-in editor, even if you re-focus the editor. But if you save the file in VS Code (without any changes) the editor will pick the modification.
